pkg/stages/steps_init.go

```diff
@@ -241,28 +241,35 @@ func GetWorkaroundsStage(sis values.System, l types.KairosLogger) []schema.Stage
 
 	if config.DefaultConfig.TrustedBoot {
 		if sis.Distro == values.Ubuntu {
-			kernel, err := getLatestKernel(l)
-			if err != nil {
-				l.Logger.Error().Msgf("Failed to get the latest kernel: %s", err)
-				return stages
+			systemVersion, err := semver.NewVersion(sis.Version)
+			if err == nil {
+				// 25.10 is the first version where this workaround is not needed
+				threshold, _ := semver.NewVersion("25.10")
+				if systemVersion.LessThan(threshold) {
+					kernel, err := getLatestKernel(l)
+					if err != nil {
+						l.Logger.Error().Msgf("Failed to get the latest kernel: %s", err)
+						return stages
+					}
+					// This looks like its out of its place as we would expect this modules to be in the initrd but this is for Trusted Boot
+					// so the initrd is creating during artifact build and contains the rootfs, so this is ok to be in here
+					stages = append(stages, []schema.Stage{
+						{
+							Name:     "Download linux-modules-extra for nvdimm modules",
+							OnlyIfOs: "Ubuntu.*",
```

Comment:
you can use `OnlyIfOsVersion` with a regex anmd yip will do it for you!

Sorry about this, I was sure this new conditionals were documented somewhere but I cant find it

Reply:
https://github.com/mudler/yip/blob/master/pkg/plugins/if_os.go#L41

Reply:
got it, I've removed the initial if conditional to check for Ubuntu also, since it already has an `OnlyIfOs`